fivetran_provider_async/utils/operator_utils.py

```diff
@@ -0,0 +1,144 @@
+    data. For more information, see the spec:
+    https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
+
+    :param config: The config dict returned from the connector API call.
+    :type config: dict
+    :param service: The service type returned from the connector or destinations
+                    API call.
+    :type service: str
+    :param schema: The schema returned from the schema API call.
+    :type schema: dict
+    :param table: The table name. Not always used.
+    :type table: str
+    """
+    if service == "gcs":
+        pattern = config["pattern"].replace("\\", "")
+        return f"{config['prefix']}{pattern}"
+    elif service == "google_sheets":
+        return config["sheet_id"]
+    elif service == "snowflake":
+        return f"{config['database']}.{schema['name_in_destination']}.{table}"
+    else:
+        # defaulting to just returning the schema name in destination for now.
+        return schema["name_in_destination"]
+
+
+def _get_openlineage_namespace(config, service, connector_id) -> str:
+    """
+    Creates an OpenLineage-compliant namespace from the relevant
+    Fivetran API data by combining the service and authority. For more information,
+    see the spec:
+    https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
+
+    :param config: The config dict returned from the connector API call.
+    :type config: dict
+    :param service: The service type returned from the connector or destinations
+                    API call.
+    :type service: str
+    """
+    if service == "snowflake":
+        name_split = config["host"].split(".")
+        return "snowflake://" + name_split[0] + "." + name_split[1]
+    elif service == "gcs":
+        return "gs://" + config["bucket"]
+    else:
+        # defaulting to this for now... better than throwing a ValueError I guess. Should really only be used when there isn't an OpenLineage spec in Naming.md.
+        return f"fivetran://{connector_id}"
+
+
+def get_dataset(
+    config,
+    service,
+    table_response,
+    column_response,
+    schema,
+    connector_id,
+    loc,
+) -> List[Dataset]:
+    datasets = []
+
+    for table_name, table in schema["tables"].items():
+        name = _get_openlineage_name(config, service, schema, table["name_in_destination"])
+        namespace = _get_openlineage_namespace(config, service, connector_id)
+        uri = f"{namespace}/{name}"
+
+        datasets.extend(
+            [
+                Dataset(
+                    namespace=namespace,
+                    name=name,
+                    facets={
+                        "SchemaDatasetFacet": _get_fields(
+                            table_id=_get_table_id(table_name, table, table_response, loc),
+                            columns=column_response,
+                            loc=loc,
+                        ),
+                        "DataSourceDatasetFacet": DataSourceDatasetFacet(name=name, uri=uri),
```

Comment:
Should this not contain input and output quality metrics? https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.md#input-dataset-facets

For example, if its a database job, it could be number of rows changed, etc

Reply:
From what I can tell, the Fivetran API does not give us this information. I'll talk to their team about if that's a possibility.